Bionic with system libs doesn't link

Steps to reproduce on a fresh install of Ubuntu Bionic:

```
export DEBIAN_FRONTEND=noninteractive && apt update && apt install curl wget htop git vim ccache -y && git clone https://github.com/bitcoin/bitcoin.git ./bitcoin-core && cd bitcoin-core && apt install build-essential libtool autotools-dev automake pkg-config bsdmainutils python3-zmq     libevent-dev libboost-system-dev libboost-filesystem-dev libboost-test-dev libboost-thread-dev  libsqlite3-dev libdb++-dev  gcc-8 g++-8 -y && ./autogen.sh && CXX=g++-8 CC=gcc-8 ./configure  --without-incompatible-bdb  && make -j $(nproc)
```

Output:

```
  CXXLD    bitcoind
libbitcoin_util.a(libbitcoin_util_a-system.o): In function `boost::system::error_category::std_category::equivalent(std::error_code const&, int) const':
/usr/include/boost/system/error_code.hpp:686: undefined reference to `boost::system::generic_category()'
/usr/include/boost/system/error_code.hpp:689: undefined reference to `boost::system::generic_category()'
/usr/include/boost/system/error_code.hpp:701: undefined reference to `boost::system::generic_category()'
libbitcoin_util.a(libbitcoin_util_a-system.o): In function `boost::system::error_category::std_category::equivalent(int, std::error_condition const&) const':
/usr/include/boost/system/error_code.hpp:656: undefined reference to `boost::system::generic_category()'
/usr/include/boost/system/error_code.hpp:659: undefined reference to `boost::system::generic_category()'
libbitcoin_util.a(libbitcoin_util_a-system.o):/usr/include/boost/system/error_code.hpp:206: more undefined references to `boost::system::generic_category()' follow
libbitcoin_util.a(libbitcoin_util_a-system.o): In function `__static_initialization_and_destruction_0':
/usr/include/boost/system/error_code.hpp:210: undefined reference to `boost::system::system_category()'
libbitcoin_util.a(libbitcoin_util_a-system.o): In function `_GLOBAL__sub_I_BITCOIN_CONF_FILENAME':
/usr/include/boost/asio/error.hpp:230: undefined reference to `boost::system::system_category()'
collect2: error: ld returned 1 exit status
Makefile:5708: recipe for target 'bitcoind' failed
make[2]: *** [bitcoind] Error 1
make[2]: Leaving directory '/bitcoin-core/src'
Makefile:16758: recipe for target 'all-recursive' failed
make[1]: *** [all-recursive] Error 1
make[1]: Leaving directory '/bitcoin-core/src'
Makefile:814: recipe for target 'all-recursive' failed
make: *** [all-recursive] Error 1
```


Initial report by @hebasto in https://github.com/bitcoin/bitcoin/pull/20744#issuecomment-1029406510

It looks like an issue exists even before std::fs.

On commit c194293883fbb656779102309b2cb3e60889feff:

```
  CXXLD    bitcoind
libbitcoin_wallet.a(libbitcoin_wallet_a-bdb.o): In function `wallet::BerkeleyDatabase::Verify(bilingual_str&)':
./src/wallet/bdb.cpp:276: undefined reference to `Db::verify(char const*, char const*, std::__1::basic_ostream<char, std::__1::char_traits<char> >*, unsigned int)'
libbitcoin_util.a(libbitcoin_util_a-system.o): In function `SetupEnvironment()':
./src/util/system.cpp:1320: undefined reference to `boost::filesystem::path::imbue(std::__1::locale const&)'
./src/util/system.cpp:1322: undefined reference to `boost::filesystem::path::imbue(std::__1::locale const&)'
clang: error: linker command failed with exit code 1 (use -v to see invocation)
Makefile:5711: recipe for target 'bitcoind' failed
make[2]: *** [bitcoind] Error 1
make[2]: Leaving directory '/bitcoin-core/src'
Makefile:16761: recipe for target 'all-recursive' failed
make[1]: *** [all-recursive] Error 1
make[1]: Leaving directory '/bitcoin-core/src'
Makefile:817: recipe for target 'all-recursive' failed
make: *** [all-recursive] Error 1

--- a/configure.ac
+++ b/configure.ac
@@ -1418,8 +1418,20 @@ if test "$use_external_signer" != "no"; then
       use_external_signer="no";
     ;;
     *)
-      use_external_signer="yes"
-      AC_DEFINE([ENABLE_EXTERNAL_SIGNER], [1], [Define if external signer support is enabled])
+      AC_MSG_CHECKING([whether Boost.Process can be used])
+      AC_LINK_IFELSE([AC_LANG_PROGRAM([[#include <boost/process.hpp>]])],
+        [have_boost_process="yes"],
+        [have_boost_process="no"])
+      AC_MSG_RESULT([$have_boost_process])
+      if test "$have_boost_process" == "yes"; then
+        use_external_signer="yes"
+        AC_DEFINE([ENABLE_EXTERNAL_SIGNER], [1], [Define if external signer support is enabled])
+      else
+        if test "$use_external_signer" = "yes"; then
+          AC_MSG_ERROR([External signing is not supported for this Boost version])
+        fi
+        use_external_signer="no";
+      fi
     ;;
   esac
 fi

